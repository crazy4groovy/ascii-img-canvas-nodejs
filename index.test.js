const target = require('.')

test('remote image, streamed', async () => {
  const imgSrc = 'https://pbs.twimg.com/profile_images/613094304094003200/5ACLHxhy_400x400.jpg'
  const result = await target(imgSrc, {height: 100, width: 100})
  expect(result).toMatchSnapshot()
})

test('remote image, streamed, htmlColor', async () => {
  const imgSrc = 'https://pbs.twimg.com/profile_images/613094304094003200/5ACLHxhy_400x400.jpg'
  const result = await target(imgSrc, {height: 100, width: 100, htmlColor: true})
  expect(result).toMatchSnapshot()
})

test('remote image, streamed, raw', async () => {
  const imgSrc = 'https://pbs.twimg.com/profile_images/613094304094003200/5ACLHxhy_400x400.jpg'
  const result = await target(imgSrc, {height: 100, width: 100, raw: true})
  expect(result).toMatchSnapshot()
})

test('local image, streamed', async () => {
  const imgSrc = './resources/cartoon-dog.jpg'
  const result = await target(imgSrc, {height: 100, width: 100})
  expect(result).toMatchSnapshot()
})

test('local image, streamed, htmlColor', async () => {
  const imgSrc = './resources/cartoon-dog.jpg'
  const result = await target(imgSrc, {height: 100, width: 100, htmlColor: true})
  expect(result).toMatchSnapshot()
})

test('local image, streamed, raw', async () => {
  const imgSrc = './resources/cartoon-dog.jpg'
  const result = await target(imgSrc, {height: 100, width: 100, raw: true})
  expect(result).toMatchSnapshot()
})
