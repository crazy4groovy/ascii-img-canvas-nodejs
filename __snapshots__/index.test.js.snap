// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image 1`] = `
"..................,,..,,,,,,,,,,,,::;;;;iiiii1111111111111111111111111111111111111111111111111111111
...,............,,,,,,,......,,,,::::;;;;i;iiiiiiiii11111111111111111111111111111111iiii111111111111
,,,,,,,...........,,,,,,.......,,,::;;::::;::;t1iiiiiiiii111111111111111111111111111ii11111111111111
,,:,,,,,,,,,,,,,,,,,,,,.....,,,,,::,:;;;111ffLfftLLff1iiii111111111111111111111111111111111111111111
:::::::::::::::::,,,,,,,,,,,,,,:::11itffLtttfffLCCCGLLL111iii111111111111111111111111111111111111111
:::::;::::::::::::::,,,,,,,,,,,ifffLLff1fLfLfGLLLCGCfLLCffft1111111111111111111111111111111111111111
::::::::,,,,,,,,,,,,,,,,,.,.,,tLLLfffttftLLLLfftfLLCGLCGGLL1f1iii11iiii1iiiiii111iii111111111111111i
::::,,,,,,.................,1tfLCLtLLffLLtLCCLfLCCCCLLfLCGGCfi11ft1i11iiiiiiiiiiiii1iiiiiiiiii11i111
,:,,,,..............;ii,.;titLfLfLCttftLLLGfCfLLLCGtffffGfCCt1tfLLftt1ii;;iiiii;iiiiiiiiiiiii1111iii
,,,,,............iii1ttf;ffLGLtLLfLLGCCLLCftLfCLfttftLLCLLCLLftftfffCft;;;;;;;;;;;iiiiiiiiiiiiiiiiii
,,,.........,:i;11ttffLLGGCGLLGGGCCftfLCCLtftL1ftLLLCLGCGfGCLffiii;itt;i;;;;;;;;;;;;;;;;;;iiiiiiiiii
..........,:tfttfCtfCLGGCtfLLtfii:fCGt1CtffLLLLCLfffCG0000CGCLGCfffftLtff;;;;;;;;;;;;;;;;;;;;;iiiiii
..........,;tGLftfCLLCGCCftfLLL1t,,;t011LCCfffCLGLLffL1C0G0GfGCitLCfCtGfLt1iti1:;;;;;;;i;;iiiiiiiiii
.......,.i1f1i1LLCGGCLLCCfLfffffffCLfLLtLCLfC0GGGGGCCGGGGLLCGGff1CGCffCLfCCfLCCftifftiiiiiiiiiiiiiii
........ifft1tiLCCfCfLttfit1GftLCLfCftLLCCGGGG0fLtfLfftCCGGCCtLGGGGLLLfftffLLLLCCCLCLti1ii11itiiiiii
.........,..;..;tffftLLL1Lt11ffftfftttGGCLG0GfftLLGCC0CLGGG0G00GGGGCL0LLLCGGGCtffGCCCLCffCLCLfttt1ii
........it;:11:;fLCLffLCC0CCtL0LtffffCCL00GCGGLG1:fCGfLLLCCGC0G000880G0G0000GGCCGCGGLLCLGCCCGCfCit1;
........;ftfLfLffLftffLCGCCCLCLLftft1CCG080LCfLCLCCG0C0GGLL00800GG08800GCG00G0GGCG00CLGCtG0GGLCLCLLL
,,.......;it1tffLGCffLfCLLGGCGCC0LLti:i,:880G80000LLC00000088CGG0G00800GGCG0GGGGGGCL00GG0CGfLC11;;i;
,,,........;LfffCGCCtLGGG1CGC000G0G00GCCC00088G88GftG0CGL0000GCG0000CG000800GGGCGCG000GfG00GGi::::::
,,,....,...tfftiLCCCf;i;LG;;i1:;t0GGLCGGC00G888L;tLLLLCGGGfCLLCGC0GGG80GGGGGGGCGCGGC0C000ff0LC1Li,,,
,,,,,,:;;;1ftfLfCCttt;;:;t8L;:1G0CLCGGG0CLGC080CLCCCC00CGCCG0000CCCLLCGG00LC00GGGGGLf08C1LtCCLGfL1L,
,,,,,ittfLLLfffftLCLLfL1iCL0:1CCCGLLLCG0GCG80@00GC0GC0GGLLL00G0GGCCfCCLG0GGGGGLLCCfL0L0000G0G0Gt,...
,,:1tt1fffCL11ffffti1t1tfft10LLCGLfCfftGCtC80080CLLCGGLLCC0tCG0CL0080GGLGGGGfLCLGGCLfLGCGtC0G0Gti:,.
,,,1fi1LtftLfLf1i;11fLfLCfLCC0CLfLCLfffftC088G8CCLLCGGLGC0G0ftC0800GCLLGGG00Gfi::iCC1fGfGfGCLLGCGC1,
,1it:ftLfLLCGLCftfff1ffLCGGC00CLLCf1CGLLCG0008G0t;L0GGG008G80L00GCGGLffLfLGGGfLftLCCLtt1i00fL0CL0GL;
;f1;i:1fLfLCCGCGG00CCCLCGLfGLfftLLCtCffCLLfLffLC8G08GG00888800GCtLCCCLLCGG0GGCCCGfLLCLffLLGf1f;;1fL1
,,,:tLfLCGGCLLCGG0fLfffCLLCLfLfLCC1LLCCCfCCLCGGCLLCGL0000GGCLLGCLGCGLfC0000880GCG0GCLLCLfffC;;;;;;;;
,,,1LfLLCffLLLCGfftffLftLGGG0CCGGGGCCffLC00GG0CCLLGfCfLLGGGGGLG0GLCCL0GGG0G00GGGG000GGGGGGGGC1LL;;;:
:i1fLftCfLttLfLCfffffLGGG00000LC8CC0CLCCGCLC00GGGLLGLCCLCCCCCLCCCLfLCCCG0CLGCCL80000G0GG0Gft::,,,,,,
i1tLt1tftLGLLCfCfLLLLftG0Gf0ffGGG0GLLCC0GCCCGCGC00CffLGCCCCLG00CCC000880CLLCGG1t00000000GGGGCft11tti
,tCi::1LffLLLffLCfLLLLLLCGG080GGLCGCLffLC00GGCGif0080CCGC0CLLLCGGCCt;11fLCCCGGCLtGGG000GGGG0CCCLLi::
;1fffLLLCfLCfCCCG0GLLGLLLCGf11t0CLLCLCCC00CG000t:8t;1;00LC0GGC0G0f1:..:tCLtL,1L,:LCL0000GG0CCLCCCLt.
itLCttCG0LCftCCfLt,:ttf;,1;,,t0LftGGifCLCCLC000LLCC;0C:f1fGGCt:........iLLLti...,ffCGG00LCG0Gfffi:..
,,:;,t,,:1LCCLfLi,,,.........ti;;:L88ftfitffLG8f1CGG;.;8@0,...............ii...:t11CLCGGCCC0fCCfL,..
,,,,,,,,:ftCii1f1;..................;88C;..iC00888C:,1@8t.....,,,,..............iCtLG0Gti1...1:....,
,,,,,............,....................f@@i..;88@@0,:G881,.......................,ii:tf.,,,,,,,,,,,,,
...................................,,,,188i,,0@880:0@@C,,,,,,,,,,,,,,,,,,,,,,,,,,,,:,,,,,,,,,,,,,,,:
........................,,,,,,,,,,,,,,,,L@8L,L@@@8888f::,,,,:::::::::::,,,,:::,,,,,,,,,,,,,,,,,,,,,,
...............,,..,,,,,,,,,,,,,,,,,,,,,,L880C888888f:::::::::::,,,,..,,,,,,,,,,,,,,,,,:::::::::::::
,,,,,,,,,,,,,,,,,,,,,,,,,,,:::::::::::::::f88888888t::::::::::::::::::::::::::::::::::::::::;;;;;;;;
,,,,,,,,,,,,,,:::::::::::::::::::::::::::::L888888L:::::::::::::::::::::::::::::::::::::::::::::::::
.,,,,,,,,,,,,,,,::::::::::::::::::::::,,,,,,L@88@8i:::::::::::::::::::::::::::::::::::::::::::::::::
..,,,,,,,,,,,,,,,,:::::::::::::::::::::,,,,,L@8888i,:::::::,::::::::::::::::::::::::::::::::::::::::
.,,,,,,,,,,,,::::::::::::::::,,,,,,,,,,,,,,;888888:,,,,,,,,,,,,:,,::,:::::::::::::::::::::::::::::::
ttttttttttfffffffffLLLLLLLLLLLLLLLLLLLLLLLLCCLLCCCCCCGGGGGGGGCCCCCCCCCCCCCCCCCCCCCCCCCCCLLLffffftttt
ttttttttttttttttttttfttttttttttttttttttttttttttttttttttttttttttttttttt1ttttt1ttttttttttttttttttttttt
fffttffttftttttttttttttttttttttttttttttttttttttttttttttttttttttttttt1tttttttttttttttt11ttttttttttttt
fftfffttttfftttttttttttttt1ttttfttttttfttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttt
ttffffftftftfftffttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttt1tttt11t
"
`;
